server/service/team_schedule.go

```diff
@@ -0,0 +1,253 @@
+	return newServer(
+		makeAuthenticatedServiceEndpoint(svc, modifyTeamScheduleEndpoint),
+		makeDecoderForTypeAndIDs(modifyTeamScheduleRequest{}, "team_id", "scheduled_query_id"),
+		opts,
+	)
+}
+
+func modifyTeamScheduleEndpoint(ctx context.Context, request interface{}, svc fleet.Service) (interface{}, error) {
+	req := request.(*modifyTeamScheduleRequest)
+	resp, err := svc.ModifyTeamScheduledQueries(ctx, req.TeamID, req.ScheduledQueryID, req.ScheduledQueryPayload)
+	if err != nil {
+		return modifyTeamScheduleResponse{Err: err}, nil
+	}
+	_ = resp
+	return modifyTeamScheduleResponse{}, nil
+}
+
+func (svc Service) ModifyTeamScheduledQueries(ctx context.Context, teamID uint, scheduledQueryID uint, query fleet.ScheduledQueryPayload) (*fleet.ScheduledQuery, error) {
+	if err := svc.authz.Authorize(ctx, &fleet.Pack{}, fleet.ActionWrite); err != nil {
+		return nil, err
+	}
+
+	gp, err := svc.ds.EnsureTeamPack(teamID)
+	if err != nil {
+		return nil, err
+	}
+
+	query.PackID = ptr.Uint(gp.ID)
+
+	return svc.ModifyScheduledQuery(ctx, scheduledQueryID, query)
+}
+
+func (mw loggingMiddleware) ModifyTeamScheduledQueries(ctx context.Context, teamID uint, scheduledQueryID uint, q fleet.ScheduledQueryPayload) (*fleet.ScheduledQuery, error) {
+	var err error
+	defer func(begin time.Time) {
+		_ = mw.loggerDebug(err).Log("method", "ModifyTeamScheduledQueries", "err", err, "took", time.Since(begin))
+	}(time.Now())
+	return mw.Service.ModifyTeamScheduledQueries(ctx, teamID, scheduledQueryID, q)
+}
+
+/////////////////////////////////////////////////////////////////////////////////
+// Delete
+/////////////////////////////////////////////////////////////////////////////////
+
+type deleteTeamScheduleRequest struct {
+	TeamID           uint `url:"team_id"`
+	ScheduledQueryID uint `url:"scheduled_query_id"`
+}
+
+type deleteTeamScheduleResponse struct {
+	Scheduled *fleet.ScheduledQuery `json:"scheduled,omitempty"`
+	Err       error                 `json:"error,omitempty"`
+}
+
+func (r deleteTeamScheduleResponse) error() error { return r.Err }
+
+func makeDeleteTeamScheduleEndpoint(svc fleet.Service, opts []kithttp.ServerOption) http.Handler {
+	return newServer(
+		makeAuthenticatedServiceEndpoint(svc, deleteTeamScheduleEndpoint),
+		makeDecoderForIDs(deleteTeamScheduleRequest{}, "team_id", "scheduled_query_id"),
+		opts,
+	)
+}
+
+func deleteTeamScheduleEndpoint(ctx context.Context, request interface{}, svc fleet.Service) (interface{}, error) {
+	req := request.(*deleteTeamScheduleRequest)
+	err := svc.DeleteTeamScheduledQueries(ctx, req.TeamID, req.ScheduledQueryID)
+	if err != nil {
+		return deleteTeamScheduleResponse{Err: err}, nil
+	}
+	return deleteTeamScheduleResponse{}, nil
+}
+
+func (svc Service) DeleteTeamScheduledQueries(ctx context.Context, teamID uint, scheduledQueryID uint) error {
+	if err := svc.authz.Authorize(ctx, &fleet.Pack{}, fleet.ActionWrite); err != nil {
+		return err
+	}
+	_ = teamID
```

Comment:
This would work for any scheduled query... maybe I missed something here. Anything we want to check?

Reply:
In theory we're just aliasing the normal delete API. I'm comfortable going ahead without a check, but if we were going to do one it would be that the query is scheduled in the matching team pack.